crates/pow/src/lib.rs

```diff
@@ -51,14 +66,14 @@ impl PoW {
     }
 
     /// Generates a new challenge.
-    pub fn build_challenge(&self, request: PowRequest) -> Challenge {
+    pub fn build_challenge(&self, account_id: AccountId, api_key: ApiKey) -> Challenge {
```

Comment:
Related to the above comment: could we make this more generic (i.e., not tied to account IDs or even API keys). For example, this could be something like:
```rust
pub fn build_challenge(&self, 
    requestor: impl Into<[u8; 32]>,
    domain: impl Into<[u8; 32]>,
) -> Challenge {
    ...
}
```
And then internally we just work with these generic notions of "requestor" and "domain". I haven't thought it through completely though - so, not sure if it would cause some unforeseen side effects.